### ISSUE TITLE
Add support for kustomize

### ISSUE THREAD

**Issue by xakraz** [segmentio/kubeapply#8]
Hi there !

I live your tool ! It is exactly what I was looking for since already a fan of Atlantis while working with Terraform.

Would it be possible to add support to kustomize.
Kustomize has a `build` command that fits in the logic of kubeapply.

Thanks for your work and considering my request.

**Comment by yolken-segment** [comment]
Yes, agreed that `kustomize` support would be nice. We don't use it internally at Segment, which is why we didn't include it in the initial version, but can definitely add it in the future!

**Comment by xakraz** [comment]
What could I do to start working on it ?
Have you any recommandations ?

**Comment by yolken-segment** [comment]
@xakraz cool, would welcome help here! The first step would be to figure out the interface- right now, we use file extensions like `.helm.yaml`, `.gotpl.yaml`, etc. to figure out how to do the expansion. Not super familiar with how `kustomize` works, but from the docs looks like this same approach will be harder for `kustomize` because there are so many different files that can be included. So, as an alternative we can have some sort of convention where we put all kustomize-related things in `kustomize` subdirectories in the profile? Open to other suggestions here if you have them.

Then, we'll need some way to specify the kustomize environment for building, maybe in the cluster config? Once that's decided, we'll need to add the code (into https://github.com/segmentio/kubeapply/blob/master/cmd/kubeapply/subcmd/expand.go#L78 and whatever libraries that's calling).

**Comment by yolken-segment** [comment]
I can help as well- I should have some free time over the next 2 weeks.

**Comment by scriptcoded** [comment]
Hey! :wave: I discovered kubeapply just today (seems pretty nice), so I might be completely misunderstanding how it actually works, but if I've understood at least the basics correct, kubeapply uses the config templates to produce an expanded configuration that's placed in the "expanded" directory. When applying the configs all kubeapply does is [call kubectl with the contents of that directory](https://github.com/segmentio/kubeapply/blob/bc99e3fb3877080879f2f77197a3d8e4d33dcca4/pkg/cluster/kube/ordered_client.go#L126-L132). If that's the case, then this might just be as simple as calling kubectl with `-k <expanded>/kustomize.yaml` instead of `-Rf <expanded>`, since kustomize is part of kubectl.

I might be way off here, just something that struck me when I realized I'd have to convert my kustomize project to plain k8s, and loose features such as config map generation :)
